flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/table/DataGeneratorConnectorITCase.java

```diff
@@ -72,4 +77,30 @@ public void testTypes() throws Exception {
 
         assertThat(results).as("Unexpected number of results").hasSize(10);
     }
+
+    @Test
+    public void testLimitPushDown() throws Exception {
+        final TestingTableEnvironment env =
+                TestingTableEnvironment.create(
+                        EnvironmentSettings.newInstance().inStreamingMode().build(),
+                        null,
+                        new TableConfig());
+
+        env.executeSql(
+                "CREATE TABLE datagen_t (\n"
+                        + "	f0 CHAR(1)\n"
+                        + ") WITH ("
+                        + "	'connector' = 'datagen'"
+                        + ")");
+
+        final Table table = env.sqlQuery("select * from datagen_t limit 5");
+        String[] explain = table.explain().split("==.*==\\s+");
+        assertThat(explain).as("Unexpected number of results").hasSize(4);
+        assertThat(explain[2])
+                .contains(
+                        "table=[[default_catalog, default_database, datagen_t, limit=[5]]], fields=[f0]");
```

Comment:
This exception is a bit hard to understand and relies on string comparison, can you at least just do the contains on the original explain string without the splitting?

Reply:
OK